File: python/code/2-plot-sequence.py
import numpy as np
import matplotlib.pyplot as plt

# Range
n = np.arange(-7, 8)

# Unit impulse function
delta_n_plus_2 = np.where(n == -2, 1, 0)
delta_n_minus_4 = np.where(n == 4, 1, 0)

# x(n) = 2δ(n+2) - δ(n-4)
x = 2 * delta_n_plus_2 - delta_n_minus_4

# Display values
print("n =", n)
print("x(n) =", x)

# Plot
plt.figure(figsize=(10, 5))
plt.stem(n, x)

plt.title(r"$x(n)=2\delta(n+2)-\delta(n-4)$")
plt.xlabel("n")
plt.ylabel("x(n)")
plt.grid(True)

plt.show()

File: matlab/code/2-plot-sequence.m
clc;
clear;
close all;

% Time index
n = -7:7;

% Unit impulse functions
delta1 = (n == -2);
delta2 = (n == 4);

% Given sequence
x = 2 * delta1 - delta2;

% Plot
stem(n, x, 'filled');
grid on;

xlabel('n');
ylabel('x[n]');
title('x[n] = 2\delta[n+2] - \delta[n-4]');

% Display values
disp('n values:');
disp(n);

disp('x[n] values:');
disp(x);